server/src/test/java/org/opensearch/extensions/action/IssueServiceAccountRequestTests.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.common.io.stream.BytesStreamInput;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class IssueServiceAccountRequestTests extends OpenSearchTestCase {
+
+    public void testIssueServiceAccountRequest() throws Exception {
+        String serviceAccountToken = "testToken";
+        IssueServiceAccountRequest request = new IssueServiceAccountRequest(serviceAccountToken);
+
+        assertEquals(serviceAccountToken, request.getServiceAccountToken());
+
+        BytesStreamOutput out = new BytesStreamOutput();
+        request.writeTo(out);
+        BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()));
+        request = new IssueServiceAccountRequest(in);
+
+        assertEquals(serviceAccountToken, request.getServiceAccountToken());
+
+        IssueServiceAccountRequest request2 = new IssueServiceAccountRequest(serviceAccountToken);
+
+        assertEquals(request, request2);
+        assertEquals(request.hashCode(), request2.hashCode());
+        assertEquals(request.toString(), "IssueServiceAccountRequest {" + "serviceAccountToken=" + serviceAccountToken + "}");
```

Comment:
(Personal preference, don't need to fix) I always thought it was silly to test exact format of a toString instead of just checking that it contains the important info.

Reply:
I will keep that in mind! Thank you :)